src/binder/utils.cpp

```diff
@@ -16,136 +16,23 @@
-        {
-#ifdef PLATFORM_UNIX
-            if (IsPathSeparator(wcChar))
-            {
-                wcChar = W('/');
-            }
 #endif // PLATFORM_UNIX
-
-            return wcChar;
-        }
-
-        inline BOOL IsDoublePathSeparator(SString::CIterator &cur)
-        {
-            return (IsPathSeparator(cur[0]) && IsPathSeparator(cur[1]));
-        }
-
-        bool NeedToRemoveDoubleAndNormalizePathSeparators(SString const &path)
-        {
-#ifdef PLATFORM_UNIX
-            return true;
-#else
-            SString::CIterator begin = path.Begin();
-            SString::CIterator end = path.End();
-            SString::CIterator cur = path.Begin();
-
-            while (cur < end)
-            {
-                if ((cur != begin) && ((cur + 2) < end) && IsDoublePathSeparator(cur))
-                {
-                    return true;
-                }
-
-                cur++;
-            }
-
-            return false;
-#endif
-        }
-
-        void RemoveDoubleAndNormalizePathSeparators(SString &path)
-        {
-            BINDER_LOG_ENTER(W("Utils::RemoveDoubleAndNormalizePathSeparators"));
-
-            SString::Iterator begin = path.Begin();
-            SString::Iterator end = path.End();
-            SString::Iterator cur = path.Begin();
-            PathString resultPath;
-
-            BINDER_LOG_STRING(W("path"), path);
-
-            while (cur < end)
-            {
-                if ((cur != begin) && ((cur + 2) < end) && IsDoublePathSeparator(cur))
-                {
-                    // Skip the doublette
-                    cur++;
-                }
-
-                resultPath.Append(ToPlatformPathSepator(cur[0]));
-                cur++;
-            }
-            
-            BINDER_LOG_STRING(W("resultPath"), resultPath);
-
-            path.Set(resultPath);
-
-            BINDER_LOG_LEAVE(W("Utils::RemoveDoubleAndNormalizePathSeparators"));
         }
     }
 
-    HRESULT FileOrDirectoryExists(PathString &path)
-    {
-        HRESULT hr = S_FALSE;
-
-        DWORD dwFileAttributes = WszGetFileAttributes(path.GetUnicode());
-        if (dwFileAttributes == INVALID_FILE_ATTRIBUTES)
-        {
-            hr = HRESULT_FROM_GetLastError();
-
-            if ((hr == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)) ||
-                (hr == HRESULT_FROM_WIN32(ERROR_PATH_NOT_FOUND)))
-            {
-                hr = S_FALSE;
-            }
-        }
-        else
-        {
-            hr = S_TRUE;
-        }
-
-        return hr;
-    }
-
-    HRESULT FileOrDirectoryExistsLog(PathString &path)
-    {
-        HRESULT hr = S_FALSE;
-        BINDER_LOG_ENTER(W("Utils::FileOrDirectoryExistsLog"));
-        BINDER_LOG_STRING(W("path"), path);
-        
-        hr = FileOrDirectoryExists(path);
-        
-        BINDER_LOG_LEAVE_HR(W("Utils::FileOrDirectoryExistsLog"), hr);
-        return hr;
-    }
-
     void MutateUrlToPath(SString &urlOrPath)
```

Comment:
Do we have any path where the actually needs this? It would be really nice to get rid of it - can be separate change.

Reply:
It's used by `AssemblyBinder::BindAssembly` - if the specified `AssemblyName` has a `CodeBase` specified this method is used to mutate it from `file:///C:\asm.dll` to `C:\asm.dll`.

There's actually quite a bit of code in the binder which handles just this special case (`AssemblyName.CodeBase` non-empty). I wonder how much is this used.

In any case I will look if it's possible to somehow merge the normal "Load from file" code paths with the "CodeBase" code paths (since they're doing more or less the same thing).

Reply:
The URL CodeBase paths should be all dead code, unreachable via public surface. If somebody specifies CodeBase on AssemblyName, it is taken care of here:

https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/src/System/Reflection/Assembly.CoreCLR.cs#L63

Reply:
Related: `Assembly.CodeBase` is being deprecated as a thing: https://github.com/dotnet/corefx/issues/41695

Reply:
"Taken care of" == "Ignored" - love this 😄 

But yay - Put it on must list of binder cleanup todos...